src/Raven.Client/Documents/Session/EntityToBlittable.cs

```diff
@@ -268,6 +268,8 @@ private static bool ShouldSimplifyJsonBasedOnType(string typeValue)
                 return true;
             if (typeValue.StartsWith("System.Collections.Generic.Dictionary`2[["))
                 return true;
+            if (typeValue.StartsWith("System.Collections.ObjectModel.Collection`1[["))
```

Comment:
No, this is a _great_ way to ensure that we'll very slowly have every collection under the sun here.
If the user has used `HashSet<string>`, what will we have here?
We need a better way to handle this.